geth/common/types.go

```diff
@@ -53,6 +53,9 @@ type NodeManager interface {
 	// Stopped node cannot be resumed, one starts a new node instead.
 	StopNode() (<-chan struct{}, error)
 
+	// Sync waits until blockchain is synchronized.
+	Sync(timeout time.Duration) error
```

Comment:
It would be nice to mention that `0` means infinite timeout and not "no timeout"?
Maybe also add that it should be > 2 seconds.

Reply:
I didn't only because it's not an implementation of the function. I'll instead add it to `./geth/node/manager.go` if that works for you as well.

Reply:
Yeah, but that means some terra incognita between a contract (an interface) and an implementation of the interface.
Maybe we could make a custom type to make the contract more explicit?
Currently, there are 2 cases that 1 parameter tries to match, using some implicit logic.
Can we "explicify" this logic using a custom type on top of `time.Duration`, so we can call
```go
m.Sync(common.SyncTimeout(1 * time.Minute))
m.Sync(common.SyncTimeoutInfinite)
```

Or just extract all the logic about that to the `main.go`.

One "go" way to do that would be to actually provide a context inside `Sync` function, so we can cancel it. Then we can just use `select` in `main.go` to provide the timeout functionality.

Reply:
@adambabik @mandrigin 
> One "go" way to do that would be to actually provide a context inside Sync function, so we can cancel it.

I think this is by far the best option here in terms of usage and contract definition. One could simply pass in `context.Background()` or `context.WithTimeout()`. That way, handling the timeout in `Sync` would be simplified and we could use the context timing logic in `main()`.